### DISCUSSION TITLE
Formatting a text file with Jupyter mode

### DISCUSSION THREAD

**Discussion by joouha** [astral-sh/ruff#9481] [Q&A] [answered]
Is there a way to tell `ruff format` to use Jupyter mode for a text file (i.e. for an `*.ipy` script containing IPython magics)?

**Comment by charliermarsh** [comment]
Ugh, unfortunately that's supported on `ruff check` but not `ruff format`. I will add it.

For example, to treat `.ipy` files as `.ipynb` files (i.e., as notebooks):

```
ruff check foo.ipy --extension=ipy:ipynb
```

**Reply by charliermarsh** [reply]
This will be fixed by https://github.com/astral-sh/ruff/pull/9483.

**Reply by joouha** [reply]
I don't want to treat the file as a notebook, but have ruff recognise the Jupyter-specific syntax in the python script:

```bash
$ cat foo.ipy
%timeit print(123)

$ ruff check foo.ipy --extension=ipy:ipynb
foo.ipy:1:1: E999 SyntaxError: Expected a Jupyter Notebook, which must be internally stored as JSON, but this file isn't valid JSON: expected value at line 1 column 1
Found 1 error.
```

**Reply by charliermarsh** [reply, marked as answer]
Ah no, that's not supported.
